From e7f358b71e0646f941417e834f7287a519883342 Mon Sep 17 00:00:00 2001
From: Michal Lazo <michal.lazo@memsource.com>
Date: Fri, 18 Jan 2019 10:14:56 +0100
Subject: [PATCH] compile with qt5.12 libjpeg

---
 .../WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Source/WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.h b/Source/WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.h
index b14e9ca7751..acf3e59d568 100644
--- a/Source/WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.h
+++ b/Source/WebCore/platform/image-decoders/jpeg/JPEGImageDecoder.h
@@ -34,6 +34,12 @@
 #undef TRUE
 #undef FALSE
 extern "C" {
+// jpeglib.h->jmorecfg.h tries to typedef int boolean; but this conflicts with
+// some Windows headers that may or may not have been included
+#ifdef HAVE_BOOLEAN
+#  undef HAVE_BOOLEAN
+#endif
+#define boolean jboolean
 #include "jpeglib.h"
 }
 
-- 
2.18.0.windows.1

