From 9b34e610312e158ab4f690a8107f8cd3dae67abe Mon Sep 17 00:00:00 2001
From: Michal Lazo <michal.lazo@memsource.com>
Date: Fri, 18 Jan 2019 10:03:07 +0100
Subject: [PATCH] disable useless features

---
 Tools/qmake/projects/run_cmake.pro | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/Tools/qmake/projects/run_cmake.pro b/Tools/qmake/projects/run_cmake.pro
index 26e4fdd58a6..65526ab4d6d 100644
--- a/Tools/qmake/projects/run_cmake.pro
+++ b/Tools/qmake/projects/run_cmake.pro
@@ -12,6 +12,23 @@ CONFIG(debug, debug|release) {
     configuration = Release
 }
 
+CMAKE_CONFIG+=ENABLE_PRINT_SUPPORT=OFF
+CMAKE_CONFIG+=ENABLE_GEOLOCATION=OFF
+CMAKE_CONFIG+=ENABLE_DEVICE_ORIENTATION=OFF
+CMAKE_CONFIG+=ENABLE_QT_WEBCHANNEL=OFF
+CMAKE_CONFIG+=ENABLE_WEBKIT2=OFF
+CMAKE_CONFIG+=USE_QT_MULTIMEDIA=OFF
+CMAKE_CONFIG+=ENABLE_TOUCH_EVENTS=OFF
+CMAKE_CONFIG+=ENABLE_VIDEO=OFF
+CMAKE_CONFIG+=ENABLE_WEB_AUDIO=OFF
+CMAKE_CONFIG+=ENABLE_SPELLCHECK=OFF
+CMAKE_CONFIG+=ENABLE_INSPECTOR_UI=ON
+CMAKE_CONFIG+=ENABLE_DRAG_SUPPORT=OFF
+CMAKE_CONFIG+=ENABLE_NETSCAPE_PLUGIN_API=OFF
+CMAKE_CONFIG+=ENABLE_SAMPLING_PROFILER=OFF
+CMAKE_CONFIG+=ENABLE_XSLT=OFF
+CMAKE_CONFIG+=ENABLE_X11_TARGET=OFF
+
 programExistsInPath(ninja): use_ninja = 1
 
 cmake_build_dir = $$system_quote($$system_path($$ROOT_BUILD_DIR/$$lower($$configuration)))
-- 
2.18.0.windows.1

