From 146e05f11c74cdcc3671a7cb3ce730b6b2a5036a Mon Sep 17 00:00:00 2001
From: Michal Lazo <michal.lazo@memsource.com>
Date: Fri, 7 Dec 2018 13:13:58 +0100
Subject: [PATCH] add qtwebkit as module

---
 .gitmodules | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/.gitmodules b/.gitmodules
index 57f4bfd..ec7c636 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -305,3 +305,10 @@
 	url = ../qtwebglplugin.git
 	branch = 5.12.0
 	status = addon
+[submodule "qtwebkit"]
+	depends = qtbase
+	recommends = qtdeclarative qtxmlpatterns
+	path = qtwebkit
+	url = ../qtwebkit.git
+	branch = 5.9
+	status = addon
-- 
2.18.0.windows.1

