diff --git a/src/core/gn_run.pro b/src/core/gn_run.pro
index 9860c454..c0d19976 100644
--- a/src/core/gn_run.pro
+++ b/src/core/gn_run.pro
@@ -46,7 +46,7 @@ build_pass|!debug_and_release {
         error("GN run error!")
     }
 
-    ninjaflags = $$(NINJAFLAGS)
+    ninjaflags = $(NINJAFLAGS)
     isEmpty(ninjaflags):!silent: ninjaflags = "-v"
 
     runninja.commands = $$NINJA $$ninjaflags -C $$gn_build_root QtWebEngineCore
