diff --git a/src/core/gn_run.pro b/src/core/gn_run.pro
index 9f6e64504..9a097d902 100644
--- a/src/core/gn_run.pro
+++ b/src/core/gn_run.pro
@@ -40,7 +40,7 @@ build_pass|!debug_and_release {
     gn_src_root = $$system_quote($$system_path($$QTWEBENGINE_ROOT/$$getChromiumSrcDir()))
     gn_python = "--script-executable=$$pythonPathForSystem()"
 
-    ninjaflags = $$(NINJAFLAGS)
+    ninjaflags = $(NINJAFLAGS)
     enableThreads = $$(GN_MORE_THREADS)
     isEmpty(enableThreads):macos {
         gn_threads = "--threads=1"
