diff --git a/src/3rdparty/chromium/url/url_canon_path.cc b/src/3rdparty/chromium/url/url_canon_path.cc
index 2e088a03fc2..eefb3e26c33 100644
--- a/src/3rdparty/chromium/url/url_canon_path.cc
+++ b/src/3rdparty/chromium/url/url_canon_path.cc
@@ -286,7 +286,7 @@ bool DoPartialPath(const CHAR* spec,
           // special case slashes. Since slashes are much more common than
           // dots, this actually increases performance measurably (though
           // slightly).
-          DCHECK(output->length() > path_begin_in_output);
+          //DCHECK(output->length() > path_begin_in_output);
           if (output->length() > path_begin_in_output &&
               output->at(output->length() - 1) == '/') {
             // Slash followed by a dot, check to see if this is means relative
-- 
2.23.0.windows.1

