diff --git a/src/3rdparty/chromium/sandbox/policy/win/sandbox_win.cc b/src/3rdparty/chromium/sandbox/policy/win/sandbox_win.cc
index b1ec529d260a9..f7b0de8ca8cab 100644
--- a/src/3rdparty/chromium/sandbox/policy/win/sandbox_win.cc
+++ b/src/3rdparty/chromium/sandbox/policy/win/sandbox_win.cc
@@ -40,6 +40,7 @@
 #include "base/win/sid.h"
 #include "base/win/win_util.h"
 #include "base/win/windows_version.h"
+#include "ppapi/buildflags/buildflags.h"
 #include "printing/buildflags/buildflags.h"
 #include "sandbox/features.h"
 #include "sandbox/policy/features.h"
@@ -1245,8 +1246,10 @@ std::string SandboxWin::GetSandboxTypeInEnglish(Sandbox sandbox_type) {
       return "Utility";
     case Sandbox::kGpu:
       return "GPU";
+#if BUILDFLAG(ENABLE_PLUGINS)
     case Sandbox::kPpapi:
       return "PPAPI";
+#endif
     case Sandbox::kNetwork:
       return "Network";
     case Sandbox::kCdm:
