From ee94ed82f016e68d3da4c17915403a5841815566 Mon Sep 17 00:00:00 2001
From: "michal.lazo" <michal.lazo@memsource.com>
Date: Fri, 9 Feb 2018 14:31:27 +0100
Subject: [PATCH] disable printsupport

---
 src/plugins/plugins.pro | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/plugins/plugins.pro b/src/plugins/plugins.pro
index 6d62420..4d8be70 100644
--- a/src/plugins/plugins.pro
+++ b/src/plugins/plugins.pro
@@ -9,5 +9,5 @@ qtHaveModule(gui) {
     !android:qtConfig(library): SUBDIRS *= generic
 }
 
-!winrt:qtHaveModule(printsupport): \
-    SUBDIRS += printsupport
+#!winrt:qtHaveModule(printsupport): \
+#    SUBDIRS += printsupport
-- 
2.9.2.windows.1
