diff --git a/src/gui/text/qsyntaxhighlighter.cpp b/src/gui/text/qsyntaxhighlighter.cpp
index cf584f6980..3207fadebd 100644
--- a/src/gui/text/qsyntaxhighlighter.cpp
+++ b/src/gui/text/qsyntaxhighlighter.cpp
@@ -72,11 +72,9 @@ public:
     inline void rehighlight(QTextCursor &cursor, QTextCursor::MoveOperation operation)
     {
         QScopedValueRollback<bool> bg(inReformatBlocks, true);
-        cursor.beginEditBlock();
         int from = cursor.position();
         cursor.movePosition(operation);
         reformatBlocks(from, 0, cursor.position() - from);
-        cursor.endEditBlock();
     }
 
     inline void _q_delayedRehighlight() {
