From 48943827924f4cd05de9e0121354b53f634557a1 Mon Sep 17 00:00:00 2001
From: "michal.lazo" <michal.lazo@memsource.com>
Date: Tue, 10 Apr 2018 16:23:33 +0200
Subject: [PATCH] fix xToCursor to better handle RTL

---
 src/gui/text/qtextlayout.cpp | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/src/gui/text/qtextlayout.cpp b/src/gui/text/qtextlayout.cpp
index cc91c10..d0b4d80 100644
--- a/src/gui/text/qtextlayout.cpp
+++ b/src/gui/text/qtextlayout.cpp
@@ -2824,24 +2824,24 @@ int QTextLine::xToCursor(qreal _x, CursorPosition cpos) const
         levels[i] = eng->layoutData->items[i+firstItem].analysis.bidiLevel;
     QTextEngine::bidiReorder(nItems, levels.data(), visualOrder.data());
 
+    bool rtl = eng->isRightToLeft();
+    bool beforeLineStart = rtl? (x >= line.textWidth): (x <= 0);
+    bool insideLine = (x > 0) && ((x < line.textWidth) || (line.justified && x < line.width));
+
     bool visual = eng->visualCursorMovement();
-    if (x <= 0) {
+    if (beforeLineStart) {
         // left of first item
-        int item = visualOrder[0]+firstItem;
+        int item = firstItem;
         QScriptItem &si = eng->layoutData->items[item];
         if (!si.num_glyphs)
             eng->shape(item);
         int pos = si.position;
-        if (si.analysis.bidiLevel % 2)
-            pos += eng->length(item);
         pos = qMax(line.from, pos);
         pos = qMin(line.from + line_length, pos);
         return pos;
-    } else if (x < line.textWidth
-               || (line.justified && x < line.width)) {
+    } else if (insideLine) {
         // has to be in one of the runs
         QFixed pos;
-        bool rtl = eng->isRightToLeft();
 
         eng->shapeLine(line);
         QVector<int> insertionPoints;
@@ -2991,13 +2991,12 @@ int QTextLine::xToCursor(qreal _x, CursorPosition cpos) const
     }
     // right of last item
 //     qDebug("right of last");
-    int item = visualOrder[nItems-1]+firstItem;
+    int item = (rtl ? visualOrder[0] + firstItem : visualOrder[nItems-1] + firstItem);
     QScriptItem &si = eng->layoutData->items[item];
     if (!si.num_glyphs)
         eng->shape(item);
     int pos = si.position;
-    if (!(si.analysis.bidiLevel % 2))
-        pos += eng->length(item);
+    pos += eng->length(item);
     pos = qMax(line.from, pos);
 
     int maxPos = line.from + line_length;
-- 
2.9.2.windows.1

