From 46f7ac5eb9175f82b9a0e271caac7c6d7e622a88 Mon Sep 17 00:00:00 2001
From: Michal Lazo <michal.lazo@memsource.com>
Date: Fri, 10 Jun 2022 10:09:00 +0200
Subject: [PATCH] ignore font 8514oem QTBUG-69634

---
 src/gui/text/windows/qwindowsfontdatabase.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gui/text/windows/qwindowsfontdatabase.cpp b/src/gui/text/windows/qwindowsfontdatabase.cpp
index 2f3fd288f2..656b52ddf8 100644
--- a/src/gui/text/windows/qwindowsfontdatabase.cpp
+++ b/src/gui/text/windows/qwindowsfontdatabase.cpp
@@ -694,6 +694,10 @@ static int QT_WIN_CALLBACK populateFontFamilies(const LOGFONT *logFont, const TE
     const wchar_t *faceNameW = f->elfLogFont.lfFaceName;
     if (faceNameW[0] && faceNameW[0] != L'@' && wcsncmp(faceNameW, L"WST_", 4)) {
         const QString faceName = QString::fromWCharArray(faceNameW);
+        // don't add bad font
+        if (faceName == QStringLiteral("8514oem"))
+            return 1;
+
         QPlatformFontDatabase::registerFontFamily(faceName);
         // Register current font's english name as alias
         const bool ttf = (textmetric->tmPitchAndFamily & TMPF_TRUETYPE);
-- 
2.26.1.windows.1

