diff --git a/src/corelib/serialization/qxmlstream.cpp b/src/corelib/serialization/qxmlstream.cpp
index 9b595000e6..8702474160 100644
--- a/src/corelib/serialization/qxmlstream.cpp
+++ b/src/corelib/serialization/qxmlstream.cpp
@@ -3435,11 +3435,11 @@ void QXmlStreamWriter::writeCDATA(QAnyStringView text)
   \note In Qt versions prior to 6.5, this function took QString, not
   QAnyStringView.
  */
-void QXmlStreamWriter::writeCharacters(QAnyStringView text)
+void QXmlStreamWriter::writeCharacters(QAnyStringView text, bool escapeWhitespaces )
 {
     Q_D(QXmlStreamWriter);
     d->finishStartElement();
-    d->writeEscaped(text);
+    d->writeEscaped(text, escapeWhitespaces );
 }
 
 
diff --git a/src/corelib/serialization/qxmlstream.h b/src/corelib/serialization/qxmlstream.h
index fffb9b689c..10d0c1c080 100644
--- a/src/corelib/serialization/qxmlstream.h
+++ b/src/corelib/serialization/qxmlstream.h
@@ -363,7 +363,7 @@ public:
     void writeTextElement(const QString &namespaceUri, const QString &name, const QString &text);
 #endif
     void writeCDATA(QAnyStringView text);
-    void writeCharacters(QAnyStringView text);
+    void writeCharacters(QAnyStringView text, bool escapeWhitespaces = false);
     void writeComment(QAnyStringView text);
 
     void writeDTD(QAnyStringView dtd);
