From dc8eee344e18c5910d7fe48a6888bb7609589476 Mon Sep 17 00:00:00 2001
From: "michal.lazo" <michal.lazo@memsource.com>
Date: Tue, 10 Apr 2018 10:30:34 +0200
Subject: [PATCH] QSyntaxHighlighter merge char formats

---
 src/gui/text/qsyntaxhighlighter.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/text/qsyntaxhighlighter.cpp b/src/gui/text/qsyntaxhighlighter.cpp
index e60d5b1..1137657 100644
--- a/src/gui/text/qsyntaxhighlighter.cpp
+++ b/src/gui/text/qsyntaxhighlighter.cpp
@@ -444,7 +444,7 @@ void QSyntaxHighlighter::setFormat(int start, int count, const QTextCharFormat &
 
     const int end = qMin(start + count, d->formatChanges.size());
     for (int i = start; i < end; ++i)
-        d->formatChanges[i] = format;
+        d->formatChanges[i].merge(format);
 }
 
 /*!
-- 
2.9.2.windows.1

