From 0048e8d9b11b4a62503bacc79c5bed24dc6731df Mon Sep 17 00:00:00 2001
From: "michal.lazo" <michal.lazo@memsource.com>
Date: Thu, 26 Jul 2018 16:28:55 +0200
Subject: [PATCH] RTL/LTR mark in blue color

---
 src/gui/text/qtextlayout.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/gui/text/qtextlayout.cpp b/src/gui/text/qtextlayout.cpp
index 3cc6e58..684ee28 100644
--- a/src/gui/text/qtextlayout.cpp
+++ b/src/gui/text/qtextlayout.cpp
@@ -2634,8 +2634,10 @@ void QTextLine::draw(QPainter *p, const QPointF &pos, const QTextLayout::FormatR
             } else if( ( eng->option.flags() & QTextOption::ShowLineAndParagraphSeparators ) &&
                      si.analysis.flags == QScriptAnalysis::BidiOperator ) {
                 QBrush c = format.foreground();
-                if( c.style() != Qt::NoBrush )
-                    p->setPen( c.color() );
+                if( c.style() != Qt::NoBrush ) {
+                    static const QColor bidiOperatorColor(23, 183, 255);
+                    p->setPen( bidiOperatorColor );
+                }
 
                 p->drawText( pos, QChar(si.analysis.bidiLevel%2? (ushort)0x21b0 : (ushort)0x21b1) );
                 p->setPen( pen );
-- 
2.9.2.windows.1

