From ad4995d702b8ab562c5dfb8afcb5cb68721f7208 Mon Sep 17 00:00:00 2001
From: "michal.lazo" <michal.lazo@memsource.com>
Date: Fri, 27 Jul 2018 11:36:21 +0200
Subject: [PATCH] RTL/LTR mark in blue color

---
 src/gui/text/qtextlayout.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/gui/text/qtextlayout.cpp b/src/gui/text/qtextlayout.cpp
index 3cc6e58..01f39fb 100644
--- a/src/gui/text/qtextlayout.cpp
+++ b/src/gui/text/qtextlayout.cpp
@@ -2633,9 +2633,8 @@ void QTextLine::draw(QPainter *p, const QPointF &pos, const QTextLayout::FormatR
                 p->setPen(pen);
             } else if( ( eng->option.flags() & QTextOption::ShowLineAndParagraphSeparators ) &&
                      si.analysis.flags == QScriptAnalysis::BidiOperator ) {
-                QBrush c = format.foreground();
-                if( c.style() != Qt::NoBrush )
-                    p->setPen( c.color() );
+                static const QColor bidiOperatorColor( 23, 183, 255 );
+                p->setPen( bidiOperatorColor );
 
                 p->drawText( pos, QChar(si.analysis.bidiLevel%2? (ushort)0x21b0 : (ushort)0x21b1) );
                 p->setPen( pen );
-- 
2.9.2.windows.1

