diff --git a/src/widgets/widgets/qcalendarwidget.cpp b/src/widgets/widgets/qcalendarwidget.cpp
index 8593001f8b..2f89214c2c 100644
--- a/src/widgets/widgets/qcalendarwidget.cpp
+++ b/src/widgets/widgets/qcalendarwidget.cpp
@@ -1159,7 +1159,7 @@ QTextCharFormat QCalendarModel::formatForCell(int row, int col) const
         format.merge(m_dateFormats.value(date));
         if(date < m_minimumDate || date > m_maximumDate)
             format.setBackground(pal.brush(cg, QPalette::Window));
-        if (m_shownMonth != date.month(m_calendar))
+        if (date < m_minimumDate || date > m_maximumDate || m_shownMonth != date.month(m_calendar))
             format.setForeground(pal.brush(QPalette::Disabled, QPalette::Text));
     }
     return format;
