From 7c98605236f9af1f7e79d41b70baa355e0d6abbe Mon Sep 17 00:00:00 2001
From: "michal.lazo" <michal.lazo@memsource.com>
Date: Tue, 11 Sep 2018 15:15:12 +0200
Subject: [PATCH] disable example notepad

---
 examples/widgets/tutorials/tutorials.pro | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/examples/widgets/tutorials/tutorials.pro b/examples/widgets/tutorials/tutorials.pro
index 0aaa119d8f..1fb49a88ba 100644
--- a/examples/widgets/tutorials/tutorials.pro
+++ b/examples/widgets/tutorials/tutorials.pro
@@ -1,2 +1,3 @@
 TEMPLATE = subdirs
-SUBDIRS +=  addressbook widgets modelview gettingStarted notepad
+SUBDIRS +=  addressbook widgets modelview gettingStarted
+#SUBDIRS += notepad
-- 
2.18.0.windows.1

