From bdf6dc4c19afcfd5babc78e92b4906b9e02a5751 Mon Sep 17 00:00:00 2001
From: Michal Lazo <michal.lazo@memsource.com>
Date: Wed, 28 Jun 2023 11:39:29 +0200
Subject: [PATCH] innovate code

---
 src/controls/Private/ScrollBar.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/controls/Private/ScrollBar.qml b/src/controls/Private/ScrollBar.qml
index eea7a737..37b096ef 100644
--- a/src/controls/Private/ScrollBar.qml
+++ b/src/controls/Private/ScrollBar.qml
@@ -138,7 +138,7 @@ Item {
             }
         }
 
-        onPressed: {
+        onPressed: (mouse) => {
             if (mouse.source !== Qt.MouseEventNotSynthesized) {
                 mouse.accepted = false
                 return
-- 
2.26.1.windows.1

