From ddc13bdc46400274f1c1d588a2bfeea168d99065 Mon Sep 17 00:00:00 2001
From: Michal Lazo <michal.lazo@memsource.com>
Date: Thu, 29 Jun 2023 16:21:23 +0200
Subject: [PATCH] set ListView width to 1 after change in QTBUG-110625

---
 src/controls/Private/TabBar.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/controls/Private/TabBar.qml b/src/controls/Private/TabBar.qml
index 1186968d..27104a32 100644
--- a/src/controls/Private/TabBar.qml
+++ b/src/controls/Private/TabBar.qml
@@ -123,7 +123,7 @@ FocusScope {
 
         // Note this will silence the binding loop warnings caused by QTBUG-35038
         // and should be removed when this issue is resolved.
-        property int contentWidthWorkaround: contentWidth > 0 ? contentWidth: 0
+        property int contentWidthWorkaround: contentWidth > 0 ? contentWidth: 1
         width: Math.min(availableWidth, count ? contentWidthWorkaround : availableWidth)
         height: currentItem ? currentItem.height : 0
 
-- 
2.26.1.windows.1

