From 06a329d2dcf07daf9f2d6d1051c78ce615edd116 Mon Sep 17 00:00:00 2001
From: Michal Lazo <michal.lazo@memsource.com>
Date: Thu, 29 Jun 2023 16:33:07 +0200
Subject: [PATCH] dont allow less then 0 max/min value

---
 src/controls/Private/qquickwheelarea.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/controls/Private/qquickwheelarea.cpp b/src/controls/Private/qquickwheelarea.cpp
index bd77ae24..dd21c943 100644
--- a/src/controls/Private/qquickwheelarea.cpp
+++ b/src/controls/Private/qquickwheelarea.cpp
@@ -141,6 +141,8 @@ void QQuickWheelArea1::wheelEvent(QWheelEvent *we)
 
 void QQuickWheelArea1::setHorizontalMinimumValue(qreal value)
 {
+    value = qMax<qreal>(0, value);
+
     if (value == m_horizontalMinimumValue)
         return;
 
@@ -155,6 +157,8 @@ qreal QQuickWheelArea1::horizontalMinimumValue() const
 
 void QQuickWheelArea1::setHorizontalMaximumValue(qreal value)
 {
+    value = qMax<qreal>(0, value);
+
     if (value == m_horizontalMaximumValue)
         return;
 
@@ -169,6 +173,8 @@ qreal QQuickWheelArea1::horizontalMaximumValue() const
 
 void QQuickWheelArea1::setVerticalMinimumValue(qreal value)
 {
+    value = qMax<qreal>(0, value);
+
     if (value == m_verticalMinimumValue)
         return;
 
@@ -183,6 +189,8 @@ qreal QQuickWheelArea1::verticalMinimumValue() const
 
 void QQuickWheelArea1::setVerticalMaximumValue(qreal value)
 {
+    value = qMax<qreal>(0, value);
+
     if (value == m_verticalMaximumValue)
         return;
 
-- 
2.26.1.windows.1

