diff --git a/src/linguist/CMakeLists.txt b/src/linguist/CMakeLists.txt
index d85254b32..44f7776f4 100644
--- a/src/linguist/CMakeLists.txt
+++ b/src/linguist/CMakeLists.txt
@@ -15,7 +15,7 @@ add_subdirectory(lrelease-pro)
 add_subdirectory(lupdate)
 add_subdirectory(lupdate-pro)
 if(QT_FEATURE_process AND QT_FEATURE_pushbutton AND QT_FEATURE_toolbutton AND TARGET Qt::Widgets AND NOT no-png)
-    add_subdirectory(linguist)
+    #add_subdirectory(linguist)
 endif()
 
 # Create a fake module that would emulate the Qt5::LinguistTools CMake Config package
